go-controller/pkg/util/multi_network.go

```diff
@@ -357,6 +378,19 @@ func (layer2NetConfInfo *Layer2NetConfInfo) Subnets() []string {
 	return subnets
 }
 
+// IPMode returns the layer2NetConfInfo's ipv4/ipv6 mode
+func (layer2NetConfInfo *Layer2NetConfInfo) IPMode() (bool, bool) {
+	var ipv6Mode, ipv4Mode bool
+	for _, subnet := range layer2NetConfInfo.ClusterSubnets {
```

Comment:
nit: would have liked to see some consolidation here with common code copied to 3 functions